crates/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -337,6 +347,29 @@ fn resolve_struct_fields(
     (generics, fields)
 }
 
+/// Create the mappings from TyAliasId -> TypeAliasType
+/// so that expressions can access the fields of structs
+fn resolve_type_aliases(
+    context: &mut Context,
+    type_aliases: HashMap<TyAliasId, UnresolvedTypeAlias>,
+    crate_id: CrateId,
+) {
+    // We must first go through the struct list once to ensure all IDs are pushed to
+    // the def_interner map. This lets structs refer to each other regardless of declaration order
+    // without resolve_struct_fields non-deterministically unwrapping a value
+    // that isn't in the HashMap.
```

Comment:
This comment was copied from `resolve_structs`, lets remove it.

It does offer some insight though: Now we have more than types than just structs being declared, but we only gather all the struct ids before resolving struct fields. We will likely need to change it to gather all struct and type aliase ids before resolving any types. Here's a possible testcase:
```rs
// struct forward referencing an alias and a struct
struct Foo { bar: Bar, baz: Baz }

// alias forward referencing a struct and an alias
type Bar = (Baz, Qux);

struct Baz { field: Field }

type Qux = Field;
```